### PR TITLE
refactor(core): Stop reporting to Sentry credential-not-found error (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
https://n8nio.sentry.io/issues/4528134408

### CODE REVIEW

File: packages/cli/src/CredentialsHelper.ts [hunk 1]
```diff
@@ -257,17 +267,17 @@ export class CredentialsHelper extends ICredentialsHelper {
 			throw new Error(`Credential "${nodeCredential.name}" of type "${type}" has no ID.`);
```
Comment (netroy) [inline packages/cli/src/CredentialsHelper.ts:267]: Does this error need to be reported, or can we change this as well?

File: packages/cli/src/CredentialsHelper.ts [hunk 2]
```diff
@@ -257,17 +267,17 @@ export class CredentialsHelper extends ICredentialsHelper {
...
-				`Credential with ID "${nodeCredential.id}" does not exist for type "${type}".`,
-			);
+		try {
+			credential = userId
+				? await Db.collections.SharedCredentials.findOneOrFail({
+						relations: ['credentials'],
+						where: { credentials: { id: nodeCredential.id, type }, userId },
+				  }).then((shared) => shared.credentials)
+				: await Db.collections.Credentials.findOneByOrFail({ id: nodeCredential.id, type });
+		} catch (error) {
```
Comment (netroy) [inline packages/cli/src/CredentialsHelper.ts:279]: Thanks for fixing this. Maybe we should look for all uses of `OrFail` and make sure that they are all wrapped in a try-catch. 👍🏽

**Review by netroy — APPROVED** [review]

File: packages/cli/src/CredentialsHelper.ts (hunk 1 above)
Comment (ivov) [reply packages/cli/src/CredentialsHelper.ts:267]: Assuming creds with `null` IDs are similar to workflows with `undefined` IDs (this shouldn't happen), then we should keep the guard reporting to Sentry, i.e. same reasoning as in https://github.com/n8n-io/n8n/pull/7535#issuecomment-1782897120

**Comment by janober** [comment]
Got released with [`n8n@1.17.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.17.0)

**Pull request merged**